development/docker-compose.yml

```diff
@@ -5,6 +5,7 @@ services:
     build:
       args:
         PYTHON_VER: "${PYTHON_VER}"
+        TARGETARCH: "${TARGETARCH}"
```

Comment:
Would have thought that `TARGETARCH` would be auto-detected by Docker - doesn't seem like something that we should require the user to specify?

Reply:
You're right. But I think I'm hitting this [bug](https://github.com/devcontainers/cli/issues/275). It seems that the automatic platform ARGs in the global scope are not recognized / detected by the `devcontainer`.

```bash
[2023-12-11T14:18:23.991Z] Start: Run: docker inspect --type image system-dev-dependencies-
[2023-12-11T14:18:24.064Z] Stop (73 ms): Run: docker inspect --type image system-dev-dependencies-
[2023-12-11T14:18:24.064Z] Path 'library/system-dev-dependencies-' for input 'docker.io/library/system-dev-dependencies-' failed validation.  Expected path to match regex '/^[a-z0-9]+([._-][a-z0-9]+)*(\/[a-z0-9]+([._-][a-z0-9]+)*)*$/'.
[2023-12-11T14:18:24.064Z] Error fetching image details: Could not parse image name 'system-dev-dependencies-'
[2023-12-11T14:18:24.065Z] Start: Run: docker pull system-dev-dependencies-
[2023-12-11T14:18:24.132Z] invalid reference format
[2023-12-11T14:18:24.133Z] Stop (68 ms): Run: docker pull system-dev-dependencies-
[2023-12-11T14:18:24.134Z] []
[2023-12-11T14:18:24.134Z] Error response from daemon: invalid reference format
```

The only workaround I found is to manually specified the environment variable.

Reply:
CI was failing, so instead of a build argument, I simply specified a default value in the dockerfile.
See: https://github.com/nautobot/nautobot/blob/20f7bf9e624d1093af996f673eedc8b97e171b05/docker/Dockerfile#L4
Tell me if it's ok with you.

Reply:
We'll need some docs for how to change this for arm64 users. There are 3 options that all require changing a git tracked file:

- update the default in `docker/Dockerfile`
- add an `ARCH: arm64` variable to the docker-compose.yml `services.nautobot.build.args`
- create a `docker-compose.override.yml` with the above build arg and add the `docker-compose.override.yml` to  your devcontainer.json `dockerComposeFile` list (preferred IMO)

Reply:
With the change we're making to the superuser creation I don't think we should include the `docker-compose.override.yml` in the `devcontainer.json` file by default

Reply:
Has been removed already.